src/components/Button/Button.vue

```diff
@@ -0,0 +1,305 @@
+<!--
+ - @copyright Copyright (c) 2020 Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @author Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @license GNU AGPL version 3 or any later version
+ -
+ - This program is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<docs>
+
+### General description
+```
+<Button
+	text="Example text" />
+<Button
+	text="Example text"
+	color="primary" />
+<Button
+	text="Example text"
+	color="success" />
+<Button
+	text="Example text"
+	color="error" />
+<Button
+	text="Example text"
+	color="warning" />
+```
+### Usage
+
+</docs>
+
+<template>
+	<button
+		v-ripple
+		class="button-vue"
+		v-bind="$attrs"
+		:class="buttonClassObject"
+		:aria-label="ariaLabel"
+		@click="handleClick"
+		@blur="handleBlur"
+		@keyup.tab.exact="handleTabUp">
+		<span class="button-vue__wrapper">
+			<span v-if="hasIcon" class="button-vue__icon">
+				<slot />
+			</span>
+			<span v-if="hasText" class="button-vue__text">{{ text }}</span>
+		</span>
+	</button>
+</template>
+<script>
+import Ripple from '../../directives/Ripple'
+
+export default {
+	name: 'Button',
+
+	directives: {
+		ripple: Ripple,
+	},
+
+	props: {
+		/**
+		 * The text of the button
+		 */
+		text: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * Specifies the button color
+		 * Accepted values: primary, error, warning, success. If left empty,
+		 * the default button style will be used.
+		 */
+		color: {
+			type: String,
+			validator(value) {
+				return ['primary', 'error', 'warning', 'success', ''].indexOf(value) !== -1
+			},
+			default: '',
+		},
+
+		/**
+		 * Specifies whether the button should span all the available width.
+		 * By default, buttons span the whole width of the container.
+		 */
+		wide: {
+			type: Boolean,
+			default: true,
+		},
+
+		/**
+		 * Always provide an aria-label to your button and try to make it more
+		 * specific than the button's title. E.g. if the title of the button is
+		 * "send" in the Mail app, the label could be "Send email"
+		 */
+		ariaLabel: {
+			type: String,
+			required: true,
+		},
```

Comment:
An aria-label is only required when the text inside the button is NOT self-explanatory.
e.g
```html
<button>Open menu</button>
<button aria-label="Close menu">X</button>
```

Reply:
Yes, I was just thinking about this... What about requiring it anyway to encourage providing a better explanation, and then remove it in the component if `===` to the title?

The reasoning behind this is that many times the visual context gives meaning to the button, and we should try to provide this context as much as possible via aria labels.

Reply:
So if no aria-label is provided it it would throw a console error that would make developers stop and think about this for a sec.

Reply:
Otherwise I fear that this will be always blank

Reply:
> What about requiring it anyway to encourage providing a better explanation

The issue is that aria-bale is not supposed to be additional data, but the data iself iirc.
If you read this with a screen reader I assume it will read the aria label, but ignoring the button content. So why even provide two then :)

I'm fine having a throw if you don't provide a content nor aria-label then :+1:

Reply:
Ok I'll try that but I remember that it I never accomplished to make a prop depend from another. In this case the required boolean would have to check another prop, the title. 
Otherwise I can perform the check in the mounted hook and throw there

Reply:
You can throw in the beforeMount if the check is not successful. But best is just letting the dev do maybe? Always a tough call between enforcing and documenting :)

Reply:
I think that what you proposed, either title or aria label, is a good policy here :)